src/internal/components/TextInputInnerAction.tsx

```diff
@@ -93,7 +93,7 @@ const TextInputAction = forwardRef<HTMLButtonElement, TextInputActionProps>(
     return (
       <Box as="span" className="TextInput-action" marginLeft={1} marginRight={1} lineHeight="0">
         {icon && !children ? (
-          <Tooltip direction={tooltipDirection} aria-label={ariaLabel}>
+          <Tooltip direction={tooltipDirection ?? 's'} text={ariaLabel ?? ''} type="label">
```

Comment:
Would this be something that we'd want to restrict in v37 so that `aria-label` is a required prop? I think this is fine since it mirrors the current behavior but in a future major release it'd be great to have this required so we don't have a non-descriptive tooltip

Reply:
>Would this be something that we'd want to restrict in v37 so that aria-label is a required prop?

I think we could. The only thing I can think of a bit of a blocker is that the `TextInput.Action` also accepts children and if children is passed, the tooltip becomes optional. (Based on just reading the code). I guess the case would be a text button as trailing action that doesn't need a tooltip. ([related additional comment](https://github.com/primer/react/pull/4097#discussion_r1433327122))

Just looking at the [current use cases at dotcom](https://musical-adventure-wlr3n3k.pages.github.io/?name=textinput.action), all seem to have icons. I wonder if it would make sense to re-visit the text button use case? 🤔 Or do we have clear uses cases for this? @mperrotti

Reply:
I don't think trailing _actions_ need to accept text, just the trailing visuals. For example: we may use trailing visual text to convey units like "minutes". See the timeout settings in https://github.com/settings/codespaces for an example.

Reply:
This is great. We can then refactor this conditional tooltip and make `aria-label` mandatory in the next major. I'll add this to the v37 issue.